### ISSUE TITLE
Plugin architecture for JPEG files

### ISSUE THREAD

**Issue by GoogleCodeExporter** [Rustemt/metadata-extractor#68]
```
Allow for different types of extraction to be turned on and off, and for 
processors defined in other jar files to be added dynamically.

A common example relates to XMP extraction. This takes a dependency upon 
Adobe's XMP jar file, which causes issues for some users.
```

Original issue reported on code.google.com by `drewnoakes` on 7 Nov 2012 at 12:50

**Comment by GoogleCodeExporter** [comment]
```
A bunch of work has already been done towards this, making it possible to 
specify what kinds of metadata processing should be performed upon a file. 
Currently this only applies to JPEG files.

It's working and available for review on the features/plugins branch.

For example, I tested extraction over 180 JPG files (from the samples repo) and 
found it to be about 40% faster if you only specify Exif processing. This also 
involves considerably less IO as only the JPEG segments required are actually 
loaded into memory (depending upon your InputStream's implementation of 'skip'.)

This kind of feature will be very visible on the surface of the API, so I want 
to think more about how it might effect it before making a release. Feedback 
very welcome.
```

Original comment by `drewnoakes` on 27 Dec 2012 at 7:30
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
Original comment by `drewnoakes` on 1 Jan 2013 at 11:48
- Added labels: **Priority-Medium**, **Type-Enhancement**
- Removed labels: **Priority-High**, **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
Changing the scope to JPEG only, as it's too complex for other file types at 
present.

...in which case, this issue is complete and will be available in 2.7.0.
```

Original comment by `drewnoakes` on 15 Jan 2013 at 8:17
- Changed title: **Plugin architecture for JPEG files**
- Changed state: **Fixed**
